readingDataPointsFilters:
#  - MaxDistDataPointsFilter:
#      maxDist: 100.0
#  - MaxDensityDataPointsFilter:
#      maxDensity: 20.0
#  - MaxPointCountDataPointsFilter:
#      maxCount: 5000
  # - FixStepSamplingDataPointsFilter:
  #     startStep: 8
  - RandomSamplingDataPointsFilter:
     prob: 0.2
  - SurfaceNormalDataPointsFilter:
      knn: 15
      epsilon: 1.33
      keepNormals: 1
      keepDensities: 1
  - MaxDensityDataPointsFilter:
      maxDensity: 50.0
  - SimpleSensorNoiseDataPointsFilter:
      sensorType: 0
  - ObservationDirectionDataPointsFilter
  - OrientNormalsDataPointsFilter 

referenceDataPointsFilters:
#  - MaxDistDataPointsFilter:
#      maxDist: 20.0
#  - MaxPointCountDataPointsFilter:
#      maxCount: 8000
#  - RandomSamplingDataPointsFilter:
#      prob: 0.5
  # - FixStepSamplingDataPointsFilter:
  #     startStep: 8
  - RandomSamplingDataPointsFilter:
     prob: 0.2
  - SurfaceNormalDataPointsFilter:
      knn: 15
      epsilon: 1.33
      keepNormals: 1
      keepDensities: 1
  - MaxDensityDataPointsFilter:
      maxDensity: 50.0
  - SimpleSensorNoiseDataPointsFilter:
      sensorType: 0
  - ObservationDirectionDataPointsFilter
  - OrientNormalsDataPointsFilter 

matcher:
  KDTreeMatcher:
    knn: 3
    maxDist: 2.0
    epsilon: 1

outlierFilters:
  - TrimmedDistOutlierFilter:
     ratio: 0.9
  - SurfaceNormalOutlierFilter:
     maxAngle: 1.5
  # - GenericDescriptorOutlierFilter:
  #    source: reference
  #    descName: probabilityStatic
  # - GenericDescriptorOutlierFilter:
  #    source: reading
  #    descName: probabilityStatic

errorMinimizer:
  PointToPlaneErrorMinimizer

transformationCheckers:
  - DifferentialTransformationChecker:
      minDiffRotErr: 0.01
      minDiffTransErr: 0.01
      smoothLength: 2
  - CounterTransformationChecker:
      maxIterationCount: 20
  - BoundTransformationChecker:
      maxRotationNorm: 0.80
      maxTranslationNorm: 10.00

inspector:
#  VTKFileInspector:
#    dumpDataLinks: 1
#    dumpReading: 1
#    dumpReference: 1
  NullInspector

logger:
#  FileLogger
  NullLogger
